### PR TITLE
Add JSON Benchmark

### PR DESCRIPTION
**Pull request by StephenBrown2**
## Change Summary

Pulling out the JSON Benchmarks from https://github.com/samuelcolvin/pydantic/pull/1291#discussion_r398499415 to be merged first and offer easier performance comparison.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: benchmarks/run.py [hunk 1]
```diff
@@ -152,28 +152,15 @@ def stdev(d):
...
-        tests += other_tests
-
-    lpad = max([len(t.package) for t in tests]) + 4
-    repeats = int(os.getenv('BENCHMARK_REPEATS', '5'))
-    print(f'testing {", ".join([t.package for t in tests])}, {repeats} times each')
+def run_tests(classes, cases, repeats, json=False):
+    if json:
+        classes = [c for c in classes if hasattr(c, 'to_json')]
+    lpad = max([len(t.package) for t in classes]) + 4
+    print(f'testing {", ".join([t.package for t in classes])}, {repeats} times each')
```
Comment (samuelcolvin) [inline benchmarks/run.py:159]:
```suggestion
    print(f'testing {", ".join(t.package for t in classes)}, {repeats} times each')
```

File: benchmarks/run.py [hunk 2]
```diff
@@ -152,28 +152,15 @@ def stdev(d):
...
-    if 'pydantic-only' not in sys.argv:
-        tests += other_tests
-
-    lpad = max([len(t.package) for t in tests]) + 4
-    repeats = int(os.getenv('BENCHMARK_REPEATS', '5'))
-    print(f'testing {", ".join([t.package for t in tests])}, {repeats} times each')
+def run_tests(classes, cases, repeats, json=False):
+    if json:
+        classes = [c for c in classes if hasattr(c, 'to_json')]
+    lpad = max([len(t.package) for t in classes]) + 4
```
Comment (samuelcolvin) [inline benchmarks/run.py:158]:
```suggestion
    lpad = max(len(t.package) for t in classes) + 4
```

File: benchmarks/run.py [hunk 3]
```diff
@@ -182,7 +169,10 @@ def main():
             test = test_class(True)
             for j in range(3):
                 for case in cases:
-                    passed, result = test.validate(case)
+                    if json:
```
Comment (samuelcolvin) [inline benchmarks/run.py:172]: i think we should try and omit the parsing/validation time from the json test.

Can we do something like

```py
if json:
    models = [m for passed, m in (test.validate(c) for c in cases) if passed]
    start = datetime.now()
    ...
        ...
        for m in models:
            test.to_json(m)
```

Comment (StephenBrown2) [reply benchmarks/run.py:172]: Certainly, I did want to exclude the parsing time as well. I wasn't sure how deep we'd want to get with that bit.

File: benchmarks/run.py (hunk 2 above)
Comment (StephenBrown2) [reply benchmarks/run.py:158]: Old iteration habits...

**Comment by StephenBrown2** [comment]
FastAPI seems to be failing its db testing test:
```
=================================== FAILURES ===================================
_______________________________ test_testing_dbs _______________________________

    def test_testing_dbs():
        # Import while creating the client to create the DB after starting the test session
        from sql_databases.sql_app.tests.test_sql_app import test_create_user
    
        test_db = Path("./test.db")
        app_db = Path("./sql_app.db")
>       test_create_user()

tests/test_tutorial/test_sql_databases/test_testing_databases.py:10: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def test_create_user():
        response = client.post(
            "/users/",
            json={"email": "deadpool@example.com", "password": "chimichangas4life"},
        )
>       assert response.status_code == 200
E       assert 400 == 200
E        +  where 400 = <Response [400]>.status_code

docs_src/sql_databases/sql_app/tests/test_sql_app.py:37: AssertionError
======================== 1 failed, 796 passed in 9.12s =========================
Makefile:129: recipe for target 'test-fastapi' failed
make: *** [test-fastapi] Error 1
##[error]Process completed with exit code 2.
```
Pretty sure that's unrelated?

**Comment by tiangolo** [comment]
Yep. FastAPI tests were failing, but they are fixed now. I also triggered the tests on this PR again after that and they are now passing again :tada: (that was actually yesterday).

**Comment by StephenBrown2** [comment]
Just a heads up, I should have some time tonight and Friday to work on finishing this up; I think I'd like to add some custom subclasses with a `Config.json_encoders` setting to handle them and compare before #1291 gets applied.

**Pull request merged**